packages/replayio/src/utils/protocol/api/ensureProcessed.ts

```diff
@@ -0,0 +1,39 @@
+import { ensureProcessedParameters, ensureProcessedResult } from "@replayio/protocol";
+import ProtocolClient from "../ProtocolClient";
+import { debug } from "../debug";
+
+export async function ensureProcessed(client: ProtocolClient, sessionId: string) {
+  await client.waitUntilAuthenticated();
+
+  let removeListeners: Function[] = [];
+
+  return new Promise<void>(async (resolve, reject) => {
+    try {
+      removeListeners.push(
+        client.listenForMessage("Recording.sessionError", (error: any) => {
+          reject(error);
+        })
```

Comment:
This should, likely, be scoped to a specific processing. From what I was able to gather here each processing gets a unique `sessionId` but we'd fail all of the pending ones based on a single `Recording.sessionError`

Reply:
Yeah, good call.

Maybe I should give up on trying to reuse a single protocol client for this. I was trying to be more efficient but I think it just adds too much complication.

Reply:
I'd stick with reusing the single protocol client. It's quite great from the efficiency PoV - it would be a shame to lose that. It's just that WS communication isn't best suited for the typical req/res cycle so we have to - at times - be cautious about this sort of thing. From what I see `error.parameters.sessionId` should be available here so we can just spy on it and reject conditionally